envbuilder.go

```diff
@@ -477,14 +477,19 @@ func Run(ctx context.Context, options Options) error {
 		if val, ok := os.LookupEnv("KANIKO_REGISTRY_MIRROR"); ok {
 			registryMirror = strings.Split(val, ";")
 		}
-		image, err := executor.DoBuild(&config.KanikoOptions{
+		var destinations []string
+		if options.CacheRepo != "" {
+			destinations = append(destinations, options.CacheRepo)
+		}
```

Comment:
Specifying `PUSH_IMAGE` without `CACHE_REPO` should fail-fast, as this is probably not what folks want.